spec/requests/spree/admin/overview_spec.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe "/admin", type: :request do
+  let(:enterprise) { create(:supplier_enterprise, name: "Feedme") }
+  let(:enterprise_user) { create(:user, enterprise_limit: 1) }
+
+  before do
+    enterprise_user.enterprise_roles.build(enterprise:).save
+    sign_in enterprise_user
+  end
+
+  describe "GET /admin" do
+    before do
+      allow(TermsOfServiceFile).to receive(:updated_at).and_return(2.hours.ago)
+    end
+
+    it "loads the dashboard page" do
+      get "/admin"
+
+      expect(response).to render_template("spree/admin/overview/single_enterprise_dashboard")
+    end
+
+    # The banner will show on all admin page, we are just testing it here
+    describe "terms of service updated banner" do
+      context "when terms of service has been updated" do
+        before { Spree::Config.enterprises_require_tos = true }
+
+        it "shows accept new ToS banner" do
+          enterprise_user.update(terms_of_service_accepted_at: nil)
+
+          get "/admin"
+
+          expect(response.body).to include("Terms of Service have been updated")
+        end
```

Comment:
This is really cool. But I wonder if it warrants a system spec. Maybe in spec/system/admin/overview_spec.rb?

Reply:
I am a bit on fence about this one, the only thing not tested really is that the button triggers the reflex properly. 
Now that I think about it, we should have a system spec, as it's an important feature.